node/src/components/block_executor.rs

```diff
@@ -98,10 +101,19 @@ pub(crate) struct BlockExecutor {
 }
 
 impl BlockExecutor {
-    pub(crate) fn new(genesis_state_root_hash: Digest, registry: Registry) -> Self {
+    pub(crate) fn new(
+        genesis_state_root_hash: Option<Digest>,
+        protocol_version: Version,
+        registry: Registry,
+    ) -> Self {
         let metrics = BlockExecutorMetrics::new(registry).unwrap();
         BlockExecutor {
             genesis_state_root_hash,
+            protocol_version: ProtocolVersion::from_parts(
```

Comment:
…and maybe `impl From<Version> for ProtocolVersion`.

Reply:
That's a problem since I believe `Version` isn't no-std compatible, and `ProtocolVersion` is in the casper-types crate which needs to support no-std.

Reply:
The impl could be `#[cfg(feature = "std")]`, could it?